sdk/core/src/services/AffinidiVaultStorageService.ts

```diff
@@ -0,0 +1,184 @@
+import { profile, KeysService } from '@affinidi/common'
+import { AffinidiVaultApiService } from '@affinidi/internal-api-clients'
+import { DidAuthService } from '@affinidi/affinidi-did-auth-lib'
+
+import { IPlatformEncryptionTools } from '../shared/interfaces'
+import { VaultCredential } from '../dto/vault.dto'
+import { isW3cCredential } from '../_helpers'
+
+type AffinidiVaultStorageOptions = {
+  accessApiKey: string
+  audienceDid: string
+  vaultUrl: string
+}
+
+@profile()
+export default class AffinidiVaultStorageService {
+  private _audienceDid: string
+  private _didAuthService: DidAuthService
+  private _keysService: KeysService
+  private _platformEncryptionTools: IPlatformEncryptionTools
+  private _vaultApiService: AffinidiVaultApiService
+
+  constructor(
+    didAuthService: DidAuthService,
+    keysService: KeysService,
+    platformEncryptionTools: IPlatformEncryptionTools,
+    options: AffinidiVaultStorageOptions,
+  ) {
+    this._audienceDid = options.audienceDid
+    this._didAuthService = didAuthService
+    this._keysService = keysService
+    this._platformEncryptionTools = platformEncryptionTools
+    this._vaultApiService = new AffinidiVaultApiService(options)
+  }
+
+  private async _authorizeVcVault(): Promise<string> {
+    const { body } = await this._vaultApiService.createDidAuthRequest({ audienceDid: this._audienceDid })
+    const responseToken = await this._didAuthService.createDidAuthResponseToken(body)
+    return responseToken
+  }
+
+  private async _encryptCredentials(credentials: any[]): Promise<VaultCredential[]> {
+    const publicKeyBuffer = this._keysService.getOwnPublicKey()
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+    const encryptedCredentials: VaultCredential[] = []
+
+    for (const credential of credentials) {
+      let credentialId = credential?.id
+      if (!isW3cCredential(credential)) {
+        credentialId = credential?.data?.id
+      }
+
+      const credentialIdHash = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+      const typeHashes = []
+      if (isW3cCredential(credential)) {
```

Comment:
I am asking because I don't know. What types of credentials do we have? As far as I understand;
- if credential is not w3c, we get credentialId from the the `credential.data.id`
- if credential is not w3c, it does not have a type array
what is the other type of credentials?

Reply:
@Yengas, we definitely should do something about that, but it's a known old problem and it's beyond the scope of this PR.

We use `isW3cCredential` throughout all SDK and cloud-wallet-api, with the assumption that it's either W3c credential (with id and types), or OADocument (with id stored in `.data.id` and without types)